# Assignment : Hypothesis Testing

In [1]:
import numpy as np
from scipy import stats

In [2]:
#State Hypothesis
# H0: Weekly operating cost follows the existing model.
# H1: Weekly operating cost is higher than the model.

- 	ˉxˉ = sample mean weekly cost (Rs. 3,050)
- 	μ = theoretical mean weekly cost according to the cost model (W = $1,000 + $5X for X = 600 units)
- 	σ = 5*25 units
- 	n = sample size (25 restaurants)

#Since:
W=1000+5X
For X = 600:
μ=1000+5(600)
μ=4000

In [3]:
x_bar = 3050
mu = 4000
sigma = 5 * 25
n = 25

In [4]:
#Calculating Test Stastistics
z = (x_bar - mu)/(sigma/np.sqrt(n))
print("Z Statistic =", z)

Z Statistic = -38.0


In [5]:
#Critical Value
alpha = 0.05
z_critical = stats.norm.ppf(0.95)
print(z_critical)

1.6448536269514722


In [6]:
#p_value
p_value = 1 - stats.norm.cdf(z)
print(p_value)

1.0


In [7]:
#Decision
if z > z_critical:
    print("Reject H0")
else:
    print("Fail to Reject H0")

Fail to Reject H0


### Conclusion
- The calculated z-statistic is much smaller than the critical value of 1.645.
- Therefore, we fail to reject the null hypothesis.
- There is insufficient statistical evidence to support the restaurant owners’ claim that weekly operating costs are higher than the theoretical model.

# Assignment: Estimation and Confidence Intervals

In [8]:
#Importing Libraries
import numpy as np
from scipy import stats

In [9]:
#Creating Dataset
data = [1.13,1.55,1.43,0.92,1.25,1.36,1.32,0.85,1.07,1.48,1.20,1.33,1.18,1.22,1.29]

In [10]:
#Sample Statistics
n = len(data)
mean = np.mean(data)
std = np.std(data, ddof=1)
print(mean)
print(std)

1.2386666666666666
0.19316412956959936


In [11]:
#99% Confidence Interval Using Sample Standard Deviation
#Sample size = 15 (<30)
#Population standard deviation(σ) unknown
#we can use t-distribution

In [12]:
#Critical Value
t_value = stats.t.ppf(0.995,df=n-1)
print(t_value)


2.976842734370834


In [13]:
#Margin of Error
ME = t_value * (std/np.sqrt(n))
print(ME)

0.14846932822817596


In [16]:
#confidence Interval
lower = mean - ME
upper = mean + ME
print("99% CI =",(lower, upper))

99% CI = (np.float64(1.0901973384384906), np.float64(1.3871359948948425))


In [17]:
#The t-distribution is used because the sample size is small (n = 15) and the population standard deviation is unknown.
#The 99% confidence interval indicates that we are 99% confident that the true mean print-head durability lies within this range.

#### If Population Standard Deviation Known- use z-Distribution


In [18]:
sigma = 0.2

In [19]:
#Critical Value
z_value = stats.norm.ppf(0.995)
print(z_value)

2.5758293035489004


In [20]:
#Margin of Error
ME = z_value * (sigma/np.sqrt(n))
print(ME)

0.13301525327090588


In [21]:
#Confidence Interval
lower = mean - ME
upper = mean + ME
print("99% CI =",(lower, upper))

99% CI = (np.float64(1.1056514133957607), np.float64(1.3716819199375725))


## Conclusion
- Using the sample standard deviation, the 99% confidence interval for mean print-head durability is approximately (1.090, 1.388) million characters.
- Using the known population standard deviation of 0.2 million characters, the 99% confidence interval is approximately (1.105, 1.373) million characters.
- The interval using the known population standard deviation is slightly narrower because the population variability is known exactly.